shell/browser/atom_browser_client.cc

```diff
@@ -988,6 +989,17 @@ bool AtomBrowserClient::PreSpawnRenderer(sandbox::TargetPolicy* policy) {
 }
 #endif  // defined(OS_WIN)
 
+bool AtomBrowserClient::IsPluginSandboxDisabled(
+    const content::PepperPluginInfo& info) {
+  const auto& unsandboxed_plugins = Browser::Get()->unsandboxed_plugins();
```

Comment:
Would it be reasonable to implement this as a callback into JS? I'm imagining an API like

```js
app.setPluginSandboxEnabledHandler((mimeType) => {
  return mimeType !== 'foo/bar'
})
```

As written, the API doesn't allow for re-enabling the sandbox after it's disabled, or for choosing which webContents are allowed to launch unsandboxed plugins. I'm not sure that it'd be likely to need such features, just asking whether we should consider them.

Reply:
I don't think you can easily map the plugin to the webContents. My thinking was that there are plugins that just don't work with sandbox at all, so we just need a way to opt-out. I don't think it's necessary to do it with more granularity.

Reply:
the JS handler on the other hand could get more information from `content::PepperPluginInfo`, including the name, path and MIME types

Reply:
@Feverqwe what do you think as the person experiencing the issue that we are trying to solve here?

Reply:
I think that any api will be fine. setPluginSandboxEnabledHandler is more flexible, but and disablePluginSandbox looks good.